archive/schema.go

```diff
@@ -0,0 +1,140 @@
+package archive
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/brimsec/zq/pkg/fs"
+)
+
+const configName = "zar.json"
+
+var DefaultConfig Config = Config{
+	Version:          0,
+	LogSizeThreshold: 500 * 1024 * 1024,
+}
+
+type Config struct {
+	Version          int `json:"version"`
+	LogSizeThreshold int `json:"log_size_threshold"`
+}
+
+func (c *Config) Write(path string) error {
+	b, err := json.Marshal(c)
+	if err != nil {
+		return err
+	}
+	f, err := ioutil.TempFile(filepath.Dir(path), "."+configName+".*")
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(b)
```

Comment:
Though not likely isn't this possible that the file doesn't write the whole buffer to the file. Consider doing `json.NewEncoder(f).Encode(c)`.

Reply:
https://pkg.go.dev/io?tab=doc#Writer says you'll get an error if the write is short, but regardless @mattnibs's suggestion is shorter than json.Marshal plus f.Write.